tests/vector/test_spaces.py

```diff
@@ -95,6 +95,32 @@ def test_batch_space(space, expected_batch_space_4):
     assert batch_space_4 == expected_batch_space_4
 
 
+def test_batch_space_seed():
+    for space in [
+        Box(0, 10, seed=123),
+        Tuple([Box(0, 5, seed=123), Box(0, 3, seed=123)], seed=123),
+        Dict(
+            {"space-1": Box(0, 5, seed=123), "space-2": Box(0, 10, seed=123)}, seed=123
+        ),
+    ]:
+        batched_space = batch_space(space)
+        assert space.np_random == batched_space.np_random
+        for _ in range(100):
+            assert np.all(space.sample() == batched_space.sample())
+
+    space = MultiDiscrete([5, 3], seed=123)
+    batched_space = batch_space(space)
+    assert space.np_random == batched_space.np_random
```

Comment:
Would perhaps be a good idea to check that the rng state isn't being replicated at other indices in the batched space, when `n > 1`. You wouldn't want the batched space to just return `n` identical samples, right?

Reply:
```
space = MultiDiscrete([5, 3], seed=123)
batched_space = batch_space(space, n=3)
print(f'{space.sample()=}')
print(f'{batched_space.sample()=}')
print(f'{space.sample()=}')
print(f'{batched_space.sample()=}')

space.sample()=array([3, 0])
batched_space.sample()=array([[3 0]
 [1 0]
 [0 2]])
space.sample()=array([1, 0])
batched_space.sample()=array([[4, 0],
       [4, 2],
       [2, 0]])
```
I dont totally understand your point, is this experiment what you mean?

Reply:
I suggest you use the `iterate` utility function (from [here](https://github.com/openai/gym/blob/e11231d84fd8b1386455709d77051addf8ded095/gym/vector/utils/spaces.py#L101)) to iterate over the batched sample, and check that all items are not the same. Something like:

```python
@pytest.mark.parametrize("space", <lots_of_diverse_spaces>)
@pytest.mark.parametrize("n", [1, 2, 5])
@pytest.mark.parametrize("base_seed", [123, 456])
def test_rng_different_at_each_index(space: Space, n: int, base_seed: int):
    space.seed(base_seed)
    batched_space = batch_space(space, n)
    batched_sample = batched_space.sample()
    slices = list(iterate(batched_space, batched_sample)
    assert not all(slice == slices[0] for slice in slices)
```

Reply:
Your idea of checking that the random number generator doesn't generate the same values for n > 1. I will add that

Reply:
New test in `tests/vector/test_spaces::test_rng_different_at_each_index`